impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java

```diff
@@ -26,19 +27,27 @@
 import java.io.IOException;
 import java.io.RandomAccessFile;
 import java.nio.channels.FileLock;
+import java.util.UUID;
 
 /**
  * @author Alex Snaps
  */
 public class DefaultLocalPersistenceService implements LocalPersistenceService {
 
+  private static final DiskStorePathManager DISK_STORE_PATH_MANAGER = new DiskStorePathManager();
+
   private final File rootDirectory;
   private FileLock lock;
   private File lockFile;
   private RandomAccessFile rw;
 
   public DefaultLocalPersistenceService(final PersistenceConfiguration persistenceConfiguration) {
-    rootDirectory = persistenceConfiguration.getRootDirectory();
+    if(persistenceConfiguration != null) {
+      rootDirectory = persistenceConfiguration.getRootDirectory();
+    } else {
+      // todo: this probably isn't smart... null shouldn't mean that!
+      rootDirectory = new File(System.getProperty("java.io.tmpdir") + UUID.randomUUID());
```

Comment:
I think it's safer to use the File(String, String) constructor rather than concatenating the path and the filename yourself.